MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -237,16 +237,16 @@ namespace WPEFramework {
             std::unique_lock<std::mutex> lck(m_callMutex);
             if (UNSOLICITED_MAINTENANCE == g_maintenance_type && internetConnectStatus){
                 LOGINFO("---------------UNSOLICITED_MAINTENANCE--------------");
-                for ( i=0;i< task_count ;i++ ){
+                for ( i=0;i< task_count && !m_abort_flag ;i++ ){
+                    LOGINFO("DBG Before Lock = %d",i);
                     task_thread.wait(lck);
+                    LOGINFO("DBG After Lock");
                     cmd=task_names_foreground[i].c_str();
                     cmd+=" &";
                     cmd+="\0";
                     m_task_map[task_names_foreground[i].c_str()]=true;
                     LOGINFO("Starting Script (USM) :  %s \n", cmd.c_str());
-                    if (!m_abort_flag){
-                        system(cmd.c_str());
-                    }
+                    system(cmd.c_str());
```

Comment:
is it expected that the script will start when m_abort_flag is true?

Reply:
I have modified it. Please check

Reply:
looks good